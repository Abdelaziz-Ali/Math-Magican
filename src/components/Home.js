function Home() {
  const style = {
    flexDirection: 'column',
    justifyContent: 'center',
    fontWeight: 'bold',
    display: 'flex',
    padding: '1rem',
  };
  return (
    <>
      <div style={style}>
        <h2>Welcome to Math Magicians!</h2>
        <p>
          Math is a broad subject that encompasses many different areas,
          including arithmetic, algebra, geometry, trigonometry, calculus, and
          statistics. It is a fundamental skill that is essential for many
          different careers and everyday tasks. Calculators are tools that can
          be used to perform mathematical calculations. They can be used to
          solve simple problems, such as addition and subtraction, or more
          complex problems, such as calculus equations.
        </p>
        <p>
          Calculators can be a helpful tool for students and professionals
          alike. They can help students learn math concepts by providing them
          with immediate feedback on their answers. They can also help
          professionals save time and effort by performing calculations quickly
          and accurately.
        </p>
        <p>
          However, it is important to remember that calculators are not a
          substitute for understanding math concepts. Students should still
          practice solving math problems by hand, as this will help them develop
          their problem-solving skills. Additionally, professionals should
          always check the accuracy of their calculator results by hand.
        </p>
      </div>
    </>
  );
}
export default Home;
