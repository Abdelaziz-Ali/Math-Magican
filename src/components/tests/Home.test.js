import renderer from 'react-test-renderer';
import Home from '../Home';

test('renders Calculator component', () => {
  const component = renderer.create(<Home />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
