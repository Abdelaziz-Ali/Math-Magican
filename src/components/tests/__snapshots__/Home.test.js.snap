// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "fontWeight": "bold",
      "justifyContent": "center",
      "padding": "1rem",
    }
  }
>
  <h2>
    Welcome to Math Magicians!
  </h2>
  <p>
    Math is a broad subject that encompasses many different areas, including arithmetic, algebra, geometry, trigonometry, calculus, and statistics. It is a fundamental skill that is essential for many different careers and everyday tasks. Calculators are tools that can be used to perform mathematical calculations. They can be used to solve simple problems, such as addition and subtraction, or more complex problems, such as calculus equations.
  </p>
  <p>
    Calculators can be a helpful tool for students and professionals alike. They can help students learn math concepts by providing them with immediate feedback on their answers. They can also help professionals save time and effort by performing calculations quickly and accurately.
  </p>
  <p>
    However, it is important to remember that calculators are not a substitute for understanding math concepts. Students should still practice solving math problems by hand, as this will help them develop their problem-solving skills. Additionally, professionals should always check the accuracy of their calculator results by hand.
  </p>
</div>
`;
